tools/common/read_svg.cpp

```diff
@@ -11,6 +13,41 @@ struct svg_layer {
     ptg_color color;
 };
 
+/*
+ * Parse number from string.
+ * @param c The string to parse.
+ * @return Rounded value.
+ */
+static long parse_number(const char*& c) {
+    // Check sign.
+    int sign = 1;
+    if (*c == '-') {
+        sign = -1;
+        ++c;
+    }
+    
+    // Numbers.
+    double n = 0.0;
+    for (; *c >= '0' && *c <= '9'; ++c) {
+        n *= 10;
+        n += *c - '0';
+    }
+
+    // Discard dot.
+    if (*c == '.')
+        ++c;
+
+    // Decimals.
+    double d = 1.0;
+    for (; (*c >= '0' && *c <= '9'); ++c) {
+        d *= 10;
+        n += (*c - '0') / d;
+    }
+    
+    // Return rounded value.
+    return static_cast<long>(sign * round(n));
```

Comment:
Subject to floating point errors? Eg. round(6.2) might not return 6.0 but 5.99999 which will be truncated to 5. Better to do (long)(n + 0.5).

Reply:
Or better, use lrint which works for negative numbers.